solutions/src/main/java/ru/sberbank/school/task09/RouteCacheService.java

```diff
@@ -0,0 +1,88 @@
+package ru.sberbank.school.task09;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import lombok.NonNull;
+import ru.sberbank.school.task09.kryo.CitySerializer;
+import ru.sberbank.school.task09.kryo.RouteSerializer;
+import ru.sberbank.school.util.Solution;
+
+import java.io.*;
+import java.time.LocalDate;
+import java.util.*;
+
+@Solution(9)
+public class RouteCacheService extends RouteService<City, Route<City>> {
+
+    private Kryo kryo;
+
+    public RouteCacheService(@NonNull String path) {
+        super(path);
+        initialize();
+    }
+
+
+    public void initialize() {
+        this.kryo = new Kryo();
+        kryo.setReferences(true);
+        kryo.register(Route.class, new RouteSerializer());
+        kryo.register(City.class, new CitySerializer());
+        kryo.register(ArrayList.class);
+        kryo.register(LinkedList.class);
+        kryo.register(LocalDate.class);
+    }
+
+
+    public void saveRoute(String key, Route route) {
+        try (OutputStream outputStream = new FileOutputStream(path + File.separator + key);
+             Output output = new Output(outputStream)) {
+            kryo.writeObject(output, route);
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
+
+    }
+
+    public Route<City> loadRoutes(String key) {
+        Route routes = null;
+        try (InputStream inputStream = new FileInputStream(path + File.separator + key);
+             Input input = new Input(inputStream)) {
```

Comment:
&#39;try resource&#39; has incorrect indentation level 13, expected level should be 16.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1668699648>*